core/libp2p/basic/readable.hpp

```diff
@@ -6,18 +6,44 @@
 #ifndef KAGOME_READABLE_HPP
 #define KAGOME_READABLE_HPP
 
-#include "libp2p/common/network_message.hpp"
+#include <functional>
+#include <outcome/outcome.hpp>
+#include <system_error>
+
+#include "common/buffer.hpp"
 
 namespace libp2p::basic {
 
-class Readable {
-  /**
-     * Read message from the connection
-     * @return optional to message, received by that connection
+  class Readable {
+   public:
+    using BufferResult = outcome::result<kagome::common::Buffer>;
+    using BufferResultCallback = void(BufferResult);
+
+    /**
+     * @brief This function is used to read data. The function call will block
+     * until exactly {@param to_read} bytes of data has been read successfully,
+     * or until an error occurs.
+     * @param to_read number of bytes to read.
+     * @return result of the operation.
+     */
+    virtual BufferResult read(uint32_t to_read) = 0;
+
+    /**
+     * @brief This function is used to read data. The function call will block
+     * until one or more bytes of data has been read successfully, or until an
+     * error occurs.
+     * @param to_read number of bytes to read.
+     * @return result of the operation.
+     */
+    virtual BufferResult readSome(uint32_t to_read) = 0;
```

Comment:
Not very clear from the docs, how the two functions work. In the first method, it reads exactly `to_read` bytes or until error happens. But in the second, why one, two... `to_read` bytes could be read until the method stops? What is the stopper in case of < to_read bytes?

Reply:
read(4) will block until all 4 bytes are received, readSome(4) will block until any number of bytes 1..4 are received. 

write(1)
readSome(4) - returns buffer or size 1

write(1)
read(4) - received 1, but blocked, waits for 3 morr

Reply:
can it be that read(4) or readSome(4) unblocked when 10 bytes received? what happens when readSome(4) unblocked and only 2 bytes received? what should we do with this?

Reply:
Any read can not return buffer of size larger than you specified